packages/react/src/utils/useTimeout.ts

```diff
@@ -0,0 +1,49 @@
+'use client';
+import { useLazyRef } from './useLazyRef';
+import { useOnMount } from './useOnMount';
+
+type TimeoutId = ReturnType<typeof setTimeout>;
```

Comment:
A similar code [caused issues in MUI X](https://github.com/mui/mui-x/pull/17533#discussion_r2058930177). Using `number` should be safer.